skimage/transform/_seam_carving.pyx

```diff
@@ -0,0 +1,214 @@
+
+    Parameters
+    ----------
+    img : (M, N) or (M, N, 3) ndarray
+        Input image whose vertical seams are to be removed.
+    iters : int
+        Number of vertical seams are to be removed.
+    energy_map : (M, N) ndarray
+        The array to decide the importance of each pixel. The higher
+        the value corresponding to a pixel, the more the algorithm will try
+        to keep it in the image.
+    num : int
+        Number of seams are to be removed.
+    border : int, optional
+        The number of pixels in the right, left and bottom end of the image
+        to be excluded from being considered for a seam. This is important as
+        certain filters just ignore image boundaries and set them to `0`.
+        By default border is set to `1`.
+
+    Returns
+    -------
+    image : (M, N - iters, 3) ndarray of float
+        The cropped image with the vertical seams removed.
+
+    References
+    ----------
+    .. [1] Shai Avidan and Ariel Shamir
+           "Seam Carving for Content-Aware Image Resizing"
+           http://www.cs.jhu.edu/~misha/ReadingSeminar/Papers/Avidan07.pdf
+    """
+    # This reference has been kept to be used for the `np.argsort` call
+    last_row_obj = np.zeros(img.shape[1], dtype=np.float)
+
+    cdef cnp.double_t[::1] last_row = last_row_obj
+    cdef Py_ssize_t[::1] sorted_indices
+    cdef cnp.uint8_t[:, ::1] seam_map = np.zeros(img.shape[0:2],
+                                                 dtype=np.uint8)
+    cdef Py_ssize_t cols = img.shape[1]
+    cdef Py_ssize_t rows = img.shape[0]
+    cdef Py_ssize_t seams_left = iters
+    cdef Py_ssize_t seams_removed
+    cdef Py_ssize_t seam_idx
+
+    cdef cnp.double_t[:, :, ::1] image = img
+    cdef cnp.int8_t[:, ::1] track_img = np.zeros(img.shape[0:2], dtype=np.int8)
+    cdef cnp.double_t[:, ::1] cumulative_img = np.zeros(img.shape[0:2],
+                                                        dtype=np.float)
+    cdef cnp.double_t[:, :, ::1] energy_img
+
+    energy_map[:, 0:border] = DBL_MAX
+    energy_map[:, cols-border:cols] = DBL_MAX
+
+    # Filters often let the boundary be `0`. If the last row is of all same
+    # values, the least value in then penultimate row will give 3 minimum
+    # enries in the last row. Hence, two successive removals will always
+    # intersect as the 3 least seams will share the same pixels except
+    # they will differ in the last row.
+    energy_map[rows-border:rows, :] = energy_map[rows-2*border:rows-border, :]
+
+    energy_map = np.ascontiguousarray(energy_map[:, :, np.newaxis])
+    energy_img = energy_map
+
+    _preprocess_image(energy_img, cumulative_img, track_img, cols)
+    last_row[...] = cumulative_img[-1, :]
+    sorted_indices = np.argsort(last_row_obj)
+    seam_idx = 0
+
+    while seams_left > 0:
+        if _mark_seam(track_img, sorted_indices[seam_idx], seam_map):
+            seams_left -= 1
+            cols -= 1
+            seam_idx += 1
+            continue
```

Comment:
Sorry I don't have time for a full review right now, but on quick examination, this `continue` is redundant.